pkg/rtc/room.go

```diff
@@ -724,6 +725,14 @@ func (r *Room) onTrackPublished(participant types.LocalParticipant, track types.
 	if r.onParticipantChanged != nil {
 		r.onParticipantChanged(participant)
 	}
+	r.lock.RUnlock()
+
+	// auto track egress
+	if egress := r.protoRoom.Egress; egress != nil && egress.Tracks != nil {
+		if err := StartTrackEgress(context.Background(), r.egressLauncher, track, egress.Tracks, r.Name(), r.ID()); err != nil {
+			r.Logger.Errorw("could not start track egress", err)
```

Comment:
thinking about the error here a bit, wondering if there's a webhook event that we should send? i.e. egress failed.